hw2/1_mini_logo.hs

```diff
@@ -1 +1,89 @@
 -- Mini Logo
+
+-- Mini Logo is a simplified version of the Logo language for programming
+-- 2D graphics. The idea behind Logo and Mini Logo is to describe simple
+-- line graphics through commands to move a pen from one position to
+-- another. The pen can either be "up" or "down". Positions are given
+-- by pairs of integers. Functions can be defined (using def) and called
+-- (using call) to reuse groups of commands. The syntax of Mini Logo is as
+-- follows
+
+-- part a: define the abstract syntax for mini logo as Cmd
+
+data Mode = Up | Down deriving (Show)
+
+data Coord = Ints (Int, Int) | Names (String, String) deriving (Show)
+
+data Pars = Pars String Pars | Name String deriving (Show)
+
+data Vals = Vals Int Vals | Int deriving (Show)
+
+data Cmd
+  = Pen Mode
+  | MoveTo Coord
+  | Def String Pars Cmd
+  | Call Vals String
```

Comment:
Probably doesn't matter, but the HW problem shows _name_ before _(vals)_ as the parameters for **call**.

Reply:
Oh, it sure is. I'll swap that out